components/MyPostsPostsList.vue

```diff
@@ -0,0 +1,154 @@
+          </template>
+        </p>
+        <div v-if="visiblePosts.length > 0">
+          <div
+            v-for="post in visiblePosts"
+            :key="'post-' + post.id"
+            class="p-0 text-start mt-1"
+          >
+            <MyMessage
+              :id="post.id"
+              :show-old="showOldPosts"
+              :expand="oldPosts.includes(post)"
+            />
+          </div>
+          <b-img
+            v-if="props.loading"
+            lazy
+            src="/loader.gif"
+            alt="Loading..."
+            width="100px"
+          />
+          <InfiniteLoading
+            :distance="scrollboxHeight"
+            @infinite="emit('load-more', $event)"
+          />
+        </div>
+        <div v-else>
+          <b-row>
+            <b-col>
+              <p>You have no active {{ props.type.toUpperCase() }}s.</p>
+            </b-col>
+          </b-row>
+          <b-row>
+            <b-col class="text-center">
+              <b-button to="/give" class="mt-1" size="lg" variant="primary">
+                <v-icon icon="gift" />&nbsp;{{ props.type }} something
+              </b-button>
+            </b-col>
+          </b-row>
+        </div>
+      </b-card-text>
+    </b-card-body>
+  </b-card>
+</template>
+
+<script setup>
+import pluralize from 'pluralize'
+import MyMessage from '~/components/MyMessage.vue'
+import InfiniteLoading from '~/components/InfiniteLoading.vue'
+
+const props = defineProps({
+  type: { type: String, required: true },
+  posts: { type: Array, required: true },
+  loading: { type: Boolean, required: true },
+  defaultExpanded: { type: Boolean, required: true },
+})
+
+const emit = defineEmits(['load-more'])
+
+const listHeaderText = computed(() => {
+  return `Your ${props.type.toUpperCase()}s`
+})
+
+const scrollboxHeight = ref(1000)
+
+const showOldPosts = ref(false)
+function toggleShowOldPosts() {
+  showOldPosts.value = !showOldPosts.value
+}
+
+// old posts are those without an outcome
+const oldPosts = computed(() => {
+  return props.posts.filter((post) => post.hasoutcome)
+})
+
+const formattedOldPostsCount = computed(() => {
+  return pluralize(
+    `old ${props.type.toUpperCase()}`,
+    oldPosts.value.length,
+    true
+  )
+})
+
+const activePosts = computed(() => {
+  return props.posts.filter((post) => !post.hasoutcome)
+})
+
+const visiblePosts = computed(() => {
+  return (
+    showOldPosts.value
+      ? activePosts.value.concat(oldPosts.value)
+      : activePosts.value
+  ).toSorted((a, b) => {
+    // promised items first, then by most recently posted
+    if (!showOldPosts.value && a.promised && !b.promised) {
+      return -1
+    } else if (!showOldPosts.value && b.promised && !a.promised) {
+      return 1
+    } else {
+      return new Date(b.arrival).getTime() - new Date(a.arrival).getTime()
+    }
+  })
+})
+</script>
+
+<style scoped>
+.restricted-height {
+  max-height: v-bind('scrollboxHeight + "px"');
```

Comment:
I don't know, you're the boss. I just took the `1000px` value from the old code. Would you like me to use `100vh` here instead?

Actually, I don't like that idea since it creates discrepancy between the value specified here in CSS and the one used for the `InfiniteScrolling` component.

Reply:
I think it would be better if `InfiniteLoading` also used `100vh`.  I don't know whether that works as a value to `rootMargin`, but if it does, let's do it in both places.

Otherwise just go with `100vh` here.